src/eoapi/vector/eoapi/vector/app.py

```diff
@@ -1,47 +1,106 @@
-"""eoAPI.vector FastAPI application."""
+"""tipg app."""
 
-from typing import Dict
+from typing import Any, List
 
-from fastapi import FastAPI
+import jinja2
+from fastapi import FastAPI, Request
 from starlette.middleware.cors import CORSMiddleware
+from starlette.templating import Jinja2Templates
+from starlette_cramjam.middleware import CompressionMiddleware
+from tipg.db import close_db_connection, connect_to_db, register_collection_catalog
+from tipg.errors import DEFAULT_STATUS_CODES, add_exception_handlers
+from tipg.factory import Endpoints as TiPgEndpoints
+from tipg.middleware import CacheControlMiddleware
+from tipg.settings import DatabaseSettings, PostgresSettings
 
+from eoapi.vector import __version__ as eoapi_vector_version
 from eoapi.vector.config import ApiSettings
-from eoapi.vector.db import close_db_connection, connect_to_db
-from eoapi.vector.factory import MVTilerFactory
-from eoapi.vector.version import __version__
 
 settings = ApiSettings()
+postgres_settings = PostgresSettings()
+db_settings = DatabaseSettings()
 
-app = FastAPI(title=settings.name, version=__version__)
+app = FastAPI(
+    title=settings.name,
+    version=eoapi_vector_version,
+    openapi_url="/api",
+    docs_url="/api.html",
+)
+
+# add eoapi_vector templates and tipg templates
+templates = Jinja2Templates(  # type: ignore
+    directory="",
+    loader=jinja2.ChoiceLoader(
+        [
+            jinja2.PackageLoader(__package__, "templates"),
+            jinja2.PackageLoader("tipg", "templates"),
+        ]
+    ),
+)
+
+# Register TiPg endpoints.
+endpoints = TiPgEndpoints(title=settings.name, templates=templates)
+app.include_router(endpoints.router, tags=["OGC API"])
 
 # Set all CORS enabled origins
 if settings.cors_origins:
     app.add_middleware(
         CORSMiddleware,
         allow_origins=settings.cors_origins,
         allow_credentials=True,
-        allow_methods=["GET", "POST", "OPTIONS"],
+        allow_methods=["GET"],
         allow_headers=["*"],
     )
 
-
-mvt = MVTilerFactory()
-app.include_router(mvt.router)
-
-
-@app.get("/healthz", description="Health Check", tags=["Health Check"])
-def ping() -> Dict:
-    """Health check."""
-    return {"ping": "pong!"}
+app.add_middleware(CacheControlMiddleware, cachecontrol=settings.cachecontrol)
+app.add_middleware(CompressionMiddleware)
+add_exception_handlers(app, DEFAULT_STATUS_CODES)
 
 
 @app.on_event("startup")
 async def startup_event() -> None:
     """Connect to database on startup."""
-    await connect_to_db(app)
+    await connect_to_db(
+        app,
+        settings=postgres_settings,
+        server_settings={"search_path": "pgstac,public", "application_name": "pgstac"},
+    )
+    await register_collection_catalog(
+        app,
+        schemas=["public"],
+        tables=db_settings.tables,
+        function_schemas=["pg_temp"],
+        # functions=db_settings.functions,
+    )
```

Comment:
One thing that is complex here is that we can't pass the custom SQL directory via variable but we will have to use `TIPG_CUSTOM_SQL_DIRECTORY` environment

https://github.com/developmentseed/tipg/blob/main/tipg/settings.py#L161-L170